### PR TITLE
Add metadata provider

### PR DESCRIPTION
**Pull request by owenherbert**
Looks like we need a metadata provider here.

**Before fix:**
```
There was 1 failure:

1) core_privacy\privacy\provider_test::test_table_coverage
The following tables with user fields must be covered with metadata providers: 
  - local_wunderbyte_table (userid)
```

**After fix:**
```
root@0609f8a7fab8:/var/www/site# vendor/bin/phpunit --testsuite core_privacy_testsuite,local_shopping_cart_testsuite,local_wunderbyte_table_testsuite,mod_booking_testsuite
Moodle 4.1.9 (Build: 20240212), 13570b66e321a92b24692d5702c3e9c20b8ed6e5
Php: 7.4.33, pgsql: 14.0 (Debian 14.0-1.pgdg110+1), OS: Linux 6.5.0-21-generic x86_64
PHPUnit 9.5.28 by Sebastian Bergmann and contributors.

.............................................................   61 / 1764 (  3%)
.............................................................  122 / 1764 (  6%)
.............................................................  183 / 1764 ( 10%)
.............................................................  244 / 1764 ( 13%)
.............................................................  305 / 1764 ( 17%)
.............................................................  366 / 1764 ( 20%)
.............................................................  427 / 1764 ( 24%)
.............................................................  488 / 1764 ( 27%)
.............................................................  549 / 1764 ( 31%)
.............................................................  610 / 1764 ( 34%)
.............................................................  671 / 1764 ( 38%)
.............................................................  732 / 1764 ( 41%)
.............................................................  793 / 1764 ( 44%)
.............................................................  854 / 1764 ( 48%)
.............................................................  915 / 1764 ( 51%)
.............................................................  976 / 1764 ( 55%)
............................................................. 1037 / 1764 ( 58%)
............................................................. 1098 / 1764 ( 62%)
............................................................. 1159 / 1764 ( 65%)
............................................................. 1220 / 1764 ( 69%)
............................................................. 1281 / 1764 ( 72%)
............................................................. 1342 / 1764 ( 76%)
............................................................. 1403 / 1764 ( 79%)
............................................................. 1464 / 1764 ( 82%)
............................................................. 1525 / 1764 ( 86%)
............................................................. 1586 / 1764 ( 89%)
............................................................. 1647 / 1764 ( 93%)
............................................................. 1708 / 1764 ( 96%)
........................................................      1764 / 1764 (100%)

Time: 01:08.247, Memory: 172.00 MB

OK (1764 tests, 12991 assertions)
```

### CODE REVIEW

File: lang/en/local_wunderbyte_table.php
Comment (owenherbert) [inline lang/en/local_wunderbyte_table.php:1]: I don't like these empty values, though the XML file isn't very descriptive...

File: classes/privacy/provider.php [hunk 1]
```diff
@@ -14,31 +14,167 @@
...
+     */
+    public static function delete_data_for_user(approved_contextlist $contextlist) {
+        global $DB;
+        $userid = $contextlist->get_user()->id;
+
+        foreach ($contextlist as $context) {
+            if ($context->contextlevel == CONTEXT_SYSTEM) {
+                foreach (self::$tables as $table) {
+                    $sql = "DELETE FROM {$table} WHERE userid = :userid";
+                    $DB->execute($sql, ['userid' => $userid]);
```
Comment (bwalkerl) [inline classes/privacy/provider.php]: Would be more appropriate to use the inbuilt $DB->delete_records('local_wunderbyte_table', ['userid' => $userid]);

File: classes/privacy/provider.php [hunk 2]
```diff
@@ -14,31 +14,167 @@
...
+     * Delete all data for all users in the specified context.
+     *
+     * @param \context $context the context to delete in.
+     */
+    public static function delete_data_for_all_users_in_context(\context $context) {
+        global $DB;
+        if ($context->contextlevel == CONTEXT_SYSTEM) {
+            foreach (self::$tables as $table) {
+                // Delete the records from table
+                $DB->delete_records($table, null);
```
Comment (bwalkerl) [inline classes/privacy/provider.php]: Null is the default second value so there's no point specifying it.

File: lang/en/local_wunderbyte_table.php
Comment (bwalkerl) [reply lang/en/local_wunderbyte_table.php:1]: I think it's best to add something even if it's just repeating the name.

**Review by bwalkerl — COMMENTED** [review]
Hey @owenherbert this is looking good, just added a few comments.

I'd also recommend creating an issue for this in the original repo if it doesn't already exist.

**Comment by bwalkerl** [comment]
Looks good to squash and merge.

**Pull request merged**